cirq-core/cirq/ops/diagonal_gate.py

```diff
@@ -144,7 +144,7 @@ def _value_equality_values_(self) -> Any:
         return tuple(self._diag_angles_radians)
 
     def _decompose_for_basis(
-        self, index: int, bit_flip: int, theta: float, qubits: Sequence['cirq.Qid']
```

Comment:
I don't think we should make the changes here since below `_decompose_` explicitly checks that `self` should not be parameterized and hence the parameter will always be a `float` here.